Create a basic quickstart guide and move the basic deployment details (#5)
submariner-io/submariner-website: src/content/quickstart/_index.en.md

+++
title = "Quickstart guide"
date = 2020-02-19T20:03:44+01:00
weight = 5
pre = "<b>1. </b>"
+++

## Basic overview

> Submariner has two main core pieces (the broker and submariner), more information about 

Reviewer: Is there a distinction between “Submariner” and “submariner” here? Are we no longer talking about the engine?

Author: I'm trying to simplify the explanation a little bit (as introduction).

"submariner" in this case would be "submariner-engine + submariner-route-agent [ + submariner global cidr ipam])

I'm sure it can be explained better ':)

this topic can be found in the [Architecture](../architecture) section.

#### The broker
The broker is an API to which all participating clusters are given access and where two objects are exchanged via CRDs:

> * Cluster(.submariner.io): defines a participating cluster and it's IP CIDRs.

Reviewer: ```suggestion
* Cluster(.submariner.io): defines a participating cluster and its IP CIDRs.
```

> * Endpoint(.submariner.io): defines a connection endpoint to a Cluster, and the reachable cluster IPs from such endpoint.

Reviewer: ```suggestion
* Endpoint(.submariner.io): defines a connection endpoint to a Cluster, and the reachable cluster IPs from the endpoint.
```

The broker must be deployed on a single Kubernetes cluster. This cluster’s API server must be reachable by all Kubernetes clusters connected by Submariner. It can be a dedicated cluster, or one of the connected clusters.

#### The submariner deployment on a cluster
Once submariner is deployed on a cluster with the proper credentials to the broker it will exchange Cluster and Endpoint objects with other clusters (via push/pull/watching), and start forming connections and routes to other clusters.

## Prerequisites

Submariner has a few requirements to get started:

- At least **2 Kubernetes** clusters, one of which is designated to serve as the central broker that is accessible by all of your connected clusters; this can be one of your connected clusters, but comes with the limitation that the cluster is required to be up to facilitate interconnectivity/negotiation.

> - Different service/pod CIDR's between clusters. This is to prevent routing conflicts.

Reviewer: ```suggestion
- Different service/pod CIDRs between clusters. This is to prevent routing conflicts.
```

+++
title = "Quickstart guide"
date = 2020-02-19T20:03:44+01:00
weight = 5
pre = "<b>1. </b>"
+++

## Basic overview

Submariner has two main core pieces (the broker and submariner), more information about 
this topic can be found in the [Architecture](../architecture) section.

#### The broker

> It's an API, to which all participating clusters are given access, and where two objects are exchanged via CRDs:

Reviewer 2: ```suggestion
The broker is an API to which all participating clusters are given access and where two objects are exchanged via CRDs:
```

* Cluster(.submariner.io): defines a participating cluster and it's IP CIDRs.
* Endpoint(.submariner.io): defines a connection endpoint to a Cluster, and the reachable cluster IPs from such endpoint.

> 

Reviewer 2: Is this Latin? Where did that come from ...

Author: It's a very old latin text used commonly in web design to fill up spaces and see how it looks: https://en.wikipedia.org/wiki/Lorem_ipsum :-)

> This part should be deployed on one K8s cluster, a participating cluster could be the broker as

Reviewer 2: ```suggestion
The broker should be deployed on one K8s cluster which can be any participating cluster as
long as all other clusters can access its API.
```

Author: Your phrasing sounds more confusing to me. I’m trying to make clear that it could be/or could be not participating on the data plane connectivity.

Reviewer 2: How about _“The broker should be deployed on one K8s cluster which can be any cluster as long as …”_

Reviewer 2: Also your phrasing seems to imply that, if it's a participating cluster, it must be accessible to all others, but it's not clear if that also applies to a non-participating cluster. Logically it should. Maybe I'm splitting hairs here but I think _"which can be any cluster as long as …"_ keeps it simple and conveys the proper meaning.

Author: May be I'm also splitting hairs, but still your original phasing also makes me think that it has to be one of the dataplane connected clusters.

Reviewer: When I find something confusing, I tend to split sentences up ;-).

The broker must be deployed on a single Kubernetes cluster. This cluster’s API server must be reachable by all Kubernetes clusters connected by Submariner. It can be a dedicated cluster, or one of the connected clusters.

Author: Thanks, I like the suggestion @skitt . Let me incorporate it, @tpantelis how does it look to you?
